evm/src/proof.rs

```diff
@@ -336,6 +346,9 @@ impl BlockMetadataTarget {
         builder.connect(bm0.block_timestamp, bm1.block_timestamp);
         builder.connect(bm0.block_number, bm1.block_number);
         builder.connect(bm0.block_difficulty, bm1.block_difficulty);
+        for i in 0..8 {
+            builder.connect(bm0.block_random[i], bm1.block_random[i]);
```

Comment:
`block_random` is connected here, but it isn't set in `set_block_metadata_target`. This leads to a `generators not run` error in a recursive proof.